### ISSUE TITLE
ranges 

### ISSUE THREAD

**Issue by JellyZhangQU** [wwwjk366/gower#15]
Hi,just wondering about the line 59

 num_ranges[col] = np.abs(1 - min / max) if (max != 0) else 0.0

is that would be   num_ranges[col] = np.abs(max-min)  ?

thanks

**Comment by 11Sadegh11** [comment]
Yeah, I was wondering about it, too. As I think, if you check lines 59 and 62, it seems the author of the code just divides the (max-min) and the Z_num by the max value so it's not wrong but I don't know what is the point of it!!!?.

Also, line code 62 will have 2 problems: 
1- if num_max is zero for a column(feature), the library will skip that column and that's wrong.
2- if num_max<0, so in line code 62, we should use abs(num_max) but in the library, the author doesn't do that.
so the library will give us the wrong results if you pass a dataset with a column (feature) whose values be totally negative or non-positive. you can check the result of gower_matrix, in order to see that the results became totally wrong, but if there's at least one positive value in that column, the library work correctly. To make it correct, I drop line 62 and modify line 59 to num_rage=(max-min) and then it works totally fine.
Hope this help.
